File: GXpert/GXpert.Web/Modules/Masters/PreAcademic/PreAcademicForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Masters.Forms;

[FormScript("Masters.PreAcademic")]
[BasedOnRow(typeof(PreAcademicRow), CheckNames = true)]
public class PreAcademicForm
{
    public string ExamName { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
    public string Thumbnail { get; set; }
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/PreAcademicForm.ts
import { StringEditor, TextAreaEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PreAcademicForm {
    ExamName: StringEditor;
    Description: TextAreaEditor;
    Thumbnail: ImageUploadEditor;
}

export class PreAcademicForm extends PrefixedContext {
    static readonly formKey = 'Masters.PreAcademic';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PreAcademicForm.init)  {
            PreAcademicForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = ImageUploadEditor;

            initFormType(PreAcademicForm, [
                'ExamName', w0,
                'Description', w1,
                'Thumbnail', w2
            ]);
        }
    }
}